cmd/memtierd/main.go

```diff
@@ -75,39 +75,62 @@ func loadConfigFile(filename string) (memtier.Policy, []memtier.Routine) {
 }
 
 func main() {
-	memtier.SetLogger(log.New(os.Stderr, "", 0))
-	optPrompt := flag.Bool("prompt", false, "launch interactive prompt (ignore other parameters)")
-	optConfig := flag.String("config", "", "launch non-interactive mode with config file")
-	optConfigDumpJSON := flag.Bool("config-dump-json", false, "dump effective configuration in JSON")
```

Comment:
since --config-dump is dropped, the related e2e tests should be modified accordingly.

- memtierd.test-suite/no-policy/n4c16/test01-mover/code.var.sh:    '-config-dump'
- memtierd.test-suite/no-policy/n4c16/test02-tracker-finder/code.var.sh:memtierd-command "tracker -create finder -config {\\\"ReportAccesses\\\":42} -config-dump -start ${MEME_PID} -counters"
- memtierd.test-suite/no-policy/n4c16/test02-tracker-finder/code.var.sh:memtierd-command "tracker -create finder -config {\\\"PagemapBitPresent\\\":false} -config-dump -start ${MEME_PID} -counters"
- memtierd.test-suite/no-policy/n4c16/test02-tracker-finder/code.var.sh:memtierd-command "tracker -create finder -config {\\\"PagemapBitPresent\\\":true} -config-dump -start ${MEME_PID} -counters"
- memtierd.test-suite/no-policy/n4c16/test02-tracker-finder/code.var.sh:memtierd-command "tracker -create finder -config \"{\\\"ReportAccesses\\\":2, \\\"PagemapBitSoftDirty\\\":true}\" -config-dump -start ${MEME_PID} -counters"

Reply:
I think we did not have test for `-config-dump` for the memtierd command line (the main function). The main program `-config-dump` dumped the configuration file that was given as `memtierd -config file.yaml`.

The `-config-dump` parameters in the tests are handled by `cmd*` functions in prompt.go. Those dumps are not affected by this PR.